### PR TITLE
Update deps and fix T4

### PR DESCRIPTION
**Pull request by MaceWindu**
- fix build warnings
- [T4] add test-cases for T4 fixes
- [T4] fix DataType handling to always fully qualify type name (previous fix doesn't take into account schemas and procedures)
- update dependencies
  - [FSharp.Core](https://www.nuget.org/packages/FSharp.Core) 4.7.2 -> 5.0.0
  - [Microsoft.AspNet.OData](https://www.nuget.org/packages/Microsoft.AspNet.OData) 7.5.0 -> 7.5.1
  - [Microsoft.Data.SQLite](https://www.nuget.org/packages/Microsoft.Data.SQLite) 3.1.8 -> 3.1.9
  - [Microsoft.Extensions.DependencyInjection](https://www.nuget.org/packages/Microsoft.Extensions.DependencyInjection) 3.1.8 -> 3.1.9
  - [Microsoft.Extensions.Logging.Abstractions](https://www.nuget.org/packages/Microsoft.Extensions.Logging.Abstractions) 3.1.8 -> 3.1.9
  - ~[MySql.Data](https://www.nuget.org/packages/MySql.Data) 8.0.21 -> 8.0.22~ reverted, v22 is even more broken than v21 (looks like testing is too expensive to afford for Oracle)
  - [Oracle.ManagedDataAccess.Core](https://www.nuget.org/packages/Oracle.ManagedDataAccess.Core) 2.19.90 -> 2.19.91
  - ~[System.Data.SQLite.Core](https://www.nuget.org/packages/System.Data.SQLite.Core) 1.0.113.1 -> 1.0.113.5~ reverted due to https://system.data.sqlite.org/index.html/tktview/eb04d8c55f685ea3a302a77a42bd902f19a50b25

### CODE REVIEW

File: Tests/Tests.T4/Databases/Access.generated.cs [hunk 1]
```diff
@@ -65,76 +64,76 @@ public TestDataDB(string configuration)
...
-		[Column("varbinaryDataType",        DbType="VARBINARY(510)", DataType=DataType.VarBinary, Length=510),               Nullable] public byte[]?   VarbinaryDataType        { get; set; } // VARBINARY(510)
-		[Column("imageDataType",            DbType="LongBinary",     DataType=DataType.Image),                               Nullable] public byte[]?   ImageDataType            { get; set; } // LongBinary
-		[Column("oleObjectDataType",        DbType="LongBinary",     DataType=DataType.Image),                               Nullable] public byte[]?   OleObjectDataType        { get; set; } // LongBinary
-		[Column("uniqueidentifierDataType", DbType="GUID",           DataType=DataType.Guid),                                Nullable] public Guid?     UniqueidentifierDataType { get; set; } // GUID
+		[Column(                            DbType="Long",           DataType=LinqToDB.DataType.Int32),                            Identity   ] public int       ID                       { get; set; } // Long
```
Comment (igor-tkachev) [inline Tests/Tests.T4/Databases/Access.generated.cs:67]: I do not think this is a good idea. What are you going to do if someone calls a table "LinqToDB", or "Table", or "Column", or "Association"?

Comment (MaceWindu) [reply Tests/Tests.T4/Databases/Access.generated.cs:67]: I agree, but this is just monkey-patching while we don't have proper solution

**Review by viceroypenguin — APPROVED** [review]

**Comment by MaceWindu** [comment]
sqlite issues caused by https://system.data.sqlite.org/index.html/tktview?name=83e8001008

**Comment by MaceWindu** [comment]
Reverting sqlite update as it actually completely broken  for linux https://system.data.sqlite.org/index.html/tktview/eb04d8c55f685ea3a302a77a42bd902f19a50b25

**Pull request merged**